cli/internal/cmd/upgradecheck.go

```diff
@@ -0,0 +1,535 @@
+
+	return cmd
+}
+
+func runUpgradeCheck(cmd *cobra.Command, args []string) error {
+	log, err := newCLILogger(cmd)
+	if err != nil {
+		return fmt.Errorf("creating logger: %w", err)
+	}
+	defer log.Sync()
+	fileHandler := file.NewHandler(afero.NewOsFs())
+	flags, err := parseUpgradeCheckFlags(cmd)
+	if err != nil {
+		return err
+	}
+	checker, err := cloudcmd.NewUpgrader(cmd.OutOrStdout(), log)
+	if err != nil {
+		return err
+	}
+	versionListFetcher := fetcher.NewFetcher()
+	rekor, err := sigstore.NewRekor()
+	if err != nil {
+		return fmt.Errorf("constructing Rekor client: %w", err)
+	}
+	up := &upgradeCheckCmd{
+		collect: &versionCollector{
+			writer:         cmd.OutOrStderr(),
+			checker:        checker,
+			verListFetcher: versionListFetcher,
+			fileHandler:    fileHandler,
+			client:         http.DefaultClient,
+			rekor:          rekor,
+			flags:          flags,
+			cliVersion:     constants.VersionInfo,
+		},
+		log: log,
+	}
+
+	return up.upgradeCheck(cmd, fileHandler, flags)
+}
+
+func parseUpgradeCheckFlags(cmd *cobra.Command) (upgradeCheckFlags, error) {
+	configPath, err := cmd.Flags().GetString("config")
+	if err != nil {
+		return upgradeCheckFlags{}, err
+	}
+	force, err := cmd.Flags().GetBool("force")
+	if err != nil {
+		return upgradeCheckFlags{}, err
+	}
+	writeConfig, err := cmd.Flags().GetBool("write-config")
+	if err != nil {
+		return upgradeCheckFlags{}, err
+	}
+	ref, err := cmd.Flags().GetString("ref")
+	if err != nil {
+		return upgradeCheckFlags{}, err
+	}
+	stream, err := cmd.Flags().GetString("stream")
+	if err != nil {
+		return upgradeCheckFlags{}, err
+	}
+	return upgradeCheckFlags{
+		configPath:   configPath,
+		force:        force,
+		writeConfig:  writeConfig,
+		ref:          ref,
+		stream:       stream,
+		cosignPubKey: constants.CosignPublicKey,
+	}, nil
+}
+
+type upgradeCheckCmd struct {
+	collect collector
+	log     debugLog
+}
+
+// upgradePlan plans an upgrade of a Constellation cluster.
+func (u *upgradeCheckCmd) upgradeCheck(cmd *cobra.Command, fileHandler file.Handler, flags upgradeCheckFlags) error {
+	conf, err := config.New(fileHandler, flags.configPath, flags.force)
+	if err != nil {
+		return config.DisplayValidationErrors(cmd.ErrOrStderr(), err)
+	}
+	u.log.Debugf("Read configuration from %q", flags.configPath)
+	// get current image version of the cluster
+	csp := conf.GetProvider()
+	u.log.Debugf("Using provider %s", csp.String())
+
+	currentServices, currentImage, currentK8s, err := u.collect.currentVersions(cmd.Context())
+	if err != nil {
+		return err
+	}
+
+	supportedServices, supportedImages, supportedK8s, err := u.collect.supportedVersions(cmd.Context(), currentImage, csp)
+	if err != nil {
+		return err
+	}
+	u.log.Debugf("current service version: %s", currentServices)
+	u.log.Debugf("supported service version: %s", supportedServices)
+	u.log.Debugf("current k8s version: %s", currentK8s)
+	u.log.Debugf("supported k8s version: %s", supportedK8s)
```

Comment:
```suggestion
	u.log.Debugf("Current service version: %s", currentServices)
	u.log.Debugf("Supported service version: %s", supportedServices)
	u.log.Debugf("Current k8s version: %s", currentK8s)
	u.log.Debugf("Supported k8s version: %s", supportedK8s)
```
Log Message should start with upper case.
Haven't checked every new log message, there might be other cases where they start in lowercase
https://github.com/edgelesssys/constellation/blob/main/dev-docs/conventions.md#logging

Reply:
Updated messages from the PR and some other places.